crates/connector_proxy/src/libs/airbyte_catalog.rs

```diff
@@ -0,0 +1,239 @@
+use std::collections::HashMap;
+
+use schemars::JsonSchema;
+use serde::ser::{SerializeStruct, Serializer};
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue;
+use validator::{Validate, ValidationError};
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum SyncMode {
+    Incremental,
+    FullRefresh,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Validate)]
+#[serde(rename_all = "snake_case")]
+pub struct Stream {
+    pub name: String,
+    pub json_schema: Box<RawValue>,
+    #[validate(length(min = 1))]
+    pub supported_sync_modes: Vec<SyncMode>,
+    pub source_defined_cursor: Option<bool>,
+    pub default_cursor_field: Option<Vec<String>>,
+    pub source_defined_primary_key: Option<Vec<Vec<String>>>,
+    pub namespace: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum DestinationSyncMode {
+    Append,
+    Overwrite,
+    AppendDedup,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Validate)]
+#[serde(rename_all = "snake_case")]
+#[validate(schema(function = "Self::validate_configured_stream"))]
+pub struct ConfiguredStream {
+    #[validate]
+    pub stream: Stream,
+    pub sync_mode: SyncMode,
+    pub destination_sync_mode: DestinationSyncMode,
+    pub cursor_field: Option<Vec<String>>,
+    pub primary_key: Option<Vec<Vec<String>>>,
+
+    // TODO: might be broken if both 'projection' and its alias is present in the JSON data.
```

Comment:
I see the pattern here -
A json object could specify two fields `field` and `estuary.dev/field`, and there values are merged.

In `serde`, `alias` was used here to specify the fields. However, in this implementation, if a json object has both fields present, it will break.
I am not very sure the usecases of these fields. If there are such usecases that needs both fields co-exist in the same object, I'll switch to a customized deserializer.